modules/services/utils/src/main/java/au/com/shiftyjelly/pocketcasts/utils/featureflag/FirebaseRemoteFeatureInitializer.kt

```diff
@@ -0,0 +1,31 @@
+package au.com.shiftyjelly.pocketcasts.utils.featureflag
+
+import au.com.shiftyjelly.pocketcasts.helper.BuildConfig
+import com.google.firebase.remoteconfig.FirebaseRemoteConfig
+import java.util.concurrent.atomic.AtomicBoolean
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlinx.coroutines.tasks.await
+import timber.log.Timber
+
+@Singleton
+class FirebaseRemoteFeatureInitializer @Inject constructor(
+    private val firebaseRemoteConfig: FirebaseRemoteConfig,
+) {
+
+    private val isInitialized = AtomicBoolean(BuildConfig.DEBUG || BuildConfig.IS_PROTOTYPE)
+
+    suspend fun initialize() {
+        if (isInitialized.get()) return
+
+        val result = try {
+            val success = firebaseRemoteConfig.fetchAndActivate().await()
+            Timber.i("Firebase feature flag refreshed")
+            success
```

Comment:
```suggestion
            withTimeout(3000L) {
                val success = firebaseRemoteConfig.fetchAndActivate().await()
                Timber.i("Firebase feature flag refreshed")
                success
            }
```
Do you think we should add a timeout in case there are any issues calling Firebase? So the user doesn't get stuck on the splash screen?